src/clincoded/static/components/hpo_term_modal.js

```diff
@@ -0,0 +1,193 @@
+            const hpoidList = _.without(checkIds, null);
+            return hpoidList;
+        }
+    },
+
+    deleteHpo(index) {
+        const hpo = this.state.hpoWithTerms.filter((hpo, hpoIndex) => {
+            return hpoIndex !== index;
+        });
+        this.setState({ hpoWithTerms: hpo });
+    },
+
+    deleteElimHpo(index) {
+        const elimHpo = this.state.hpoElimWithTerms.filter((hpo, hpoIndex) => {
+            return hpoIndex !== index;
+        });
+        this.setState({ hpoElimWithTerms: elimHpo });
+    },
+    
+    /**
+     * Method to fetch HPO term names and append them to HPO ids
+    */
+    fetchHpoName() {
+        let hpoIds = this.refs['hpoid'].getValue();
+        const hpoidList = this.validateHpo(hpoIds);
+        if (hpoidList) {
+            hpoidList.forEach(id => {
+                const url = external_url_map['HPOApi'] + id;
+                // Make the OLS REST API call
+                this.getRestData(url).then(response => {
+                    const termLabel = response['details']['name'];
+                    const hpoWithTerm = `${termLabel} (${id})`;
+                    this.setState({ hpoWithTerms: [...this.state.hpoWithTerms, hpoWithTerm] });
+                }).catch(err => {
+                    // Unsuccessful retrieval
+                    console.warn('Error in fetching HPO data =: %o', err);
+                    const hpoWithTerm = id + ' (note: term not found)';
+                    this.setState({ hpoWithTerms: [...this.state.hpoWithTerms, hpoWithTerm] });
+                });
+            });
+        }
+    },
+
+    /**
+     * Method to fetch HPO term names and append them to HPO In Elimination ids
+     */
+    fetchHpoInElimName() {
+        let hpoIds = this.refs['hpoid'].getValue();
+        const hpoidList = this.validateHpo(hpoIds);
+        if (hpoidList) {
+            hpoidList.forEach(id => {
+                const url = external_url_map['HPOApi'] + id;
+                // Make the OLS REST API call
+                this.getRestData(url).then(response => {
+                    const termLabel = response['details']['name'];
+                    const hpoWithTerm = `${termLabel} (${id})`;
+                    this.setState({ hpoElimWithTerms: [...this.state.hpoElimWithTerms, hpoWithTerm] });
+                }).catch(err => {
+                    // Unsuccessful retrieval
+                    console.warn('Error in fetching HPO data =: %o', err);
+                    const hpoWithTerm = id + ' (note: term not found)';
+                    this.setState({ hpoElimWithTerms: [...this.state.hpoElimWithTerms, hpoWithTerm] });
+                });
+            });
+        }
+    },
+
+    render() {
+        let hpoWithTerms = this.state.hpoWithTerms ? this.state.hpoWithTerms : [];
+        let hpoElimWithTerms = this.state.hpoElimWithTerms ? this.state.hpoElimWithTerms : [];
+
```

Comment:
Suggestion: if want to display list without duplicated hpo, remove duplicated ones from lists.
        let hpoWithTerms = _.uniq(this.state.hpoWithTerms ? this.state.hpoWithTerms : []);
        let hpoElimWithTerms = _.uniq(this.state.hpoElimWithTerms ? this.state.hpoElimWithTerms : []);